packages/next/src/server/app-render/action-encryption.ts

```diff
@@ -0,0 +1,119 @@
+/* eslint-disable import/no-extraneous-dependencies */
+import 'server-only'
+
+/* eslint-disable import/no-extraneous-dependencies */
+import {
+  renderToReadableStream,
+  decodeReply,
+} from 'react-server-dom-webpack/server.edge'
+/* eslint-disable import/no-extraneous-dependencies */
+import {
+  createFromReadableStream,
+  encodeReply,
+} from 'react-server-dom-webpack/client.edge'
+
+import { streamToString } from '../stream-utils/node-web-streams-helper'
+import {
+  arrayBufferToString,
+  decrypt,
+  encrypt,
+  getActionEncryptionKey,
+  getClientReferenceManifestSingleton,
+  getServerModuleMap,
+  stringToUint8Array,
+} from './action-encryption-utils'
+
+async function decodeActionBoundArg(actionId: string, arg: string) {
+  const key = await getActionEncryptionKey(actionId)
+  if (typeof key === 'undefined') {
+    if (process.env.NODE_ENV === 'production') {
+      throw new Error(
+        `Missing encryption key in production. This is a bug in Next.js`
+      )
+    }
+
+    // No encryption needed during development.
+    return arg
```

Comment:
Elsewhere it seems like in dev we expect to have a key that is shared for all actions. Generally I'd say that having a failure mode that is prod only is not desirable. If we require the key in dev we can unify this across environment however it doesn't really tell us something all that interesting since the dev key is shared.

Reply:
Yes I forgot the remove that - fixed now!